apiserver/getcachedvhdimages.go

```diff
@@ -0,0 +1,52 @@
+package apiserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	agent "github.com/Azure/agentbaker/pkg/agent"
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+)
+
+const (
+	// RoutePathGetCachedVersionsOnVHD the route path to get cached vhd images.
+	RoutePathGetCachedVersionsOnVHD string = "/getcachedversionsonvhd"
+)
+
+type CachedOnVHD struct {
+	CachedFromManifest   map[string]datamodel.ProcessedManifest `json:"cached_from_manifest"`
```

Comment:
nit: I think we usually go for camelCase when defining JSON keys

Reply:
We actually need to keep these capitalized so that they are properly exported. Looking through other files, seems like the JSON keys are all capitalized too.

Reply:
err I meant the json tag

e.g. `cached_from_manifest` -> `cachedFromManifest`

and yes you must always export fields which are tagged with JSON so the json unmarshaller can actually access those fields, otherwise you'll get zero-values everywhere (I totally haven't made this mistake numerous times).